### PR TITLE
Work branch

### PR DESCRIPTION
**Pull request by Hipst0man**

### CODE REVIEW

File: main.cpp [hunk 1]
```diff
@@ -1 +1,313 @@
...
+#include <stdlib.h>
+#include <string.h>
+
+
+// Initializes list
+// void StringListInit(char*** list); 
+void StringListInit(char*** list)
+{
+    if (*list) return;
+    *list = (char**)malloc(2 * sizeof(char*));
```
Comment (Stepbura) [inline main.cpp:11]: Please use calloc instead of malloc

File: main.cpp [hunk 2]
```diff
@@ -1 +1,313 @@
...
+    (*list)[0] = nullptr;
+    (*list)[1] = nullptr;
+}
+
+// Destroy list and set pointer to NULL.
+// void StringListDestroy(char*** list);
+void StringListDestroy(char*** list)
+{
+    if (!*list) return;
+    char** current(*list), **previous(nullptr);
```
Comment (Stepbura) [inline main.cpp:21]: Please use NULL instead of nullptr

File: main.cpp [hunk 3]
```diff
@@ -1 +1,313 @@
...
+    
+    *list = nullptr;
+    list[0] = nullptr;
+}
+
+// Inserts value at the end of the list.
+// void StringListAdd(char*** list, String str);
+void StringListAdd(char*** head, const char* val)
+{
+    if (!val[0]) return;
```
Comment (Stepbura) [inline main.cpp:41]: Please add verification if val is not NULL

File: main.cpp [hunk 4]
```diff
@@ -1 +1,313 @@
...
+
+    if (!(*head)[0] && !(*head)[1])
+    {
+        (*head)[0] = (char*)malloc(strlen(val)+1);
+        if ((*head)[0]) strcpy((*head)[0], val);
```
Comment (Stepbura) [inline main.cpp:46]: Please use strcpy_s instead of strcpy

File: main.cpp [hunk 5]
```diff
@@ -1 +1,313 @@
...
+    }
+}
+
+//Additional
+//Removes node from tail
+void popBack(char*** head)
+{
+    if(!*head) return;
+
+    char** current(*head), **previous(nullptr);
```
Comment (Stepbura) [inline main.cpp:95]: Use NULL, not nullptr

File: main.cpp (hunk 4 above)
Comment (Hipst0man) [reply main.cpp:46]: In VSC i don't have strcpy_s, this func is only available in Visual Studio

File: main.cpp [hunk 6]
```diff
@@ -1 +1,313 @@
...
+        {
+            char* tmp = (char*)realloc(current[0], strlen(after)+1);
+
+            if (tmp)
+            {
+                current[0] = tmp;
+                strcpy(current[0], after);
+            }
+            
+            else exit(1);
```
Comment (Stepbura) [inline main.cpp:229]: Why do you need exit(1) here? Use return or break

File: main.cpp (hunk 4 above)
Comment (Stepbura) [reply main.cpp:46]: Yes, correct. it requires to add in header <string.h>

**Pull request closed without merging**